proxy/src/main/java/com/wavefront/agent/listeners/tracing/HeartbeatMetricKey.java

```diff
@@ -15,17 +17,20 @@
   @Nonnull
   private final String cluster;
   @Nonnull
-  private String shard;
+  private final String shard;
+  @Nonnull
+  private final String source;
   @Nonnull
-  private String source;
+  private final Map<String, String> customTags;
 
   public HeartbeatMetricKey(String application, String service, String cluster, String shard,
-                            String source) {
+                            String source, Map<String, String> customTags) {
```

Comment:
If we have these properties annotated as @Nonnull, we should do the same for the constructor arguments